packages/material-ui-unstyled/src/Unstable_TrapFocus/Unstable_TrapFocus.js

```diff
@@ -338,7 +316,7 @@ function Unstable_TrapFocus(props) {
   };
 
   const handleFocusSentinel = (event) => {
-    if (!activated.current) {
+    if (nodeToRestore.current === null) {
```

Comment:
This handler wasn't covered by tests. The behavior of this handler is now covered in `test/e2e/fixtures/Unstable_TrapFocus/DefaultOpenLazyTrapFocus.tsx`